widget/simple_renderer.go

```diff
@@ -0,0 +1,57 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+)
+
+// newSimpleRenderer creates a new simpleRenderer.
+func newSimpleRenderer(widget SimpleWidget, objects []fyne.CanvasObject, layout func(size fyne.Size)) *simpleRenderer {
+	return &simpleRenderer{
+		widget:  widget,
+		objects: objects,
+		layout:  layout,
+	}
+}
+
+// simpleRenderer is a renderer providing the basic rendering functionality used
+// by SimpleWidgetBase.
+type simpleRenderer struct {
+	widget  fyne.Widget
+	objects []fyne.CanvasObject
+
+	layout func(size fyne.Size)
+}
+
+// MinSize returns the minimum size of the widget that is rendered by this renderer.
+func (s *simpleRenderer) MinSize() fyne.Size {
+	return fyne.Size{Width: 0, Height: 0}
+}
+
+// Refresh is a hook that is called if the widget has updated and needs to be redrawn.
+// This might trigger a Layout.
+func (s *simpleRenderer) Refresh() {
+	s.Layout(s.widget.Size())
+}
```

Comment:
Are there use cases where this is not enough? Should we be able to overwrite the `Refresh` function? If those cases are rare, I'd say for those cases one can fall back to implementing the `fyne.Widget` the usual way.

Reply:
`Refresh` doesn't have to call `Layout` in most cases, this could reduce performance if it's the default behaviour. (However, I definitely don't mind the idea of a `RefreshAfterLayout` sort of option, if you ever decide to go down the options route.)

Reply:
So, if I understand correctly, `Refresh` is called for e.g. data updates and `Layout` needs to be called for resizing/repositioning? Or what exactly is the definition/distinction?

Since the goal is to simplify widget implementation, it kinda feels complicated to me to have to understand the difference and implement 3 different functions (CreateRenderer, Layout and Refresh). I'd like to keep it with 2 (CreateRenderer and Layout). Or do you see another option?
I'm not sure I got the `RefreshAfterLayout` option. Don't see how that would help keeping it simple.

Reply:
Does Layout trigger a rebuild all the way down the widget tree even if there is no position/size change?

Reply:
Refresh can now be overwritten and will default to calling Layout.

Reply:
> So, if I understand correctly, `Refresh` is called for e.g. data updates and `Layout` needs to be called for resizing/repositioning? Or what exactly is the definition/distinction?

That is the destinction. Layout is very expensive as it requires checking the size of all child objects.
Refresh is faster because it changes properties and pushes the rest of the work to renderer/GPU.